scheduler/scheduler-server/src/test/java/functionaltests/monitor/SchedulerMonitorsHandler.java

```diff
@@ -461,6 +475,29 @@ private void waitWithMonitor(EventMonitor monitor, long timeout) throws ProActiv
         throw new ProActiveTimeoutException("timeout elapsed");
     }
 
+    public void waitWithAnyMonitor(long timeout, EventMonitor... monitors) throws ProActiveTimeoutException {
+        TimeoutAccounter counter = TimeoutAccounter.getAccounter(timeout);
+        synchronized (monitors) {
+            Arrays.stream(monitors).forEach(monitor -> monitor.setTimeouted(false));
+
+            while (!counter.isTimeoutElapsed()) {
+                if (Arrays.stream(monitors).anyMatch(monitor -> monitor.eventOccured)) {
+                    return;
+                }
+
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException e) {
+                    System.exit(1);
```

Comment:
That's a bit hardcore, why not propagate the interrupt ?

Reply:
Roger that